tests/models/patchtsmixer/test_modeling_patchtsmixer.py

```diff
@@ -0,0 +1,1076 @@
+        else:
+            self.assertEqual(output.prediction_logits.shape, target_output.shape)
+
+        self.assertEqual(output.last_hidden_state.shape, enc_output.shape)
+
+        if output_hidden_states is True:
+            self.assertEqual(len(output.hidden_states), params["num_layers"])
+
+        else:
+            self.assertEqual(output.hidden_states, None)
+
+        self.assertEqual(output.loss.item() < 100, True)
+
+        if config.loss == "nll" and task in ["forecast", "regression"]:
+            samples = mdl.generate(self.__class__.data)
+            self.assertEqual(samples.sequences.shape, ref_samples.shape)
+
+    def test_forecast(self):
+        for mode in ["common_channel", "mix_channel"]:
+            for self_attn in [True, False]:
+                for scaling in [True, False, "mean", "std"]:
+                    for gated_attn in [True, False]:
+                        for forecast_channel_indices in [None, [0, 2]]:
+                            for loss in ["mse", "nll"]:
+                                params = self.__class__.params.copy()
+                                params.update(
+                                    mode=mode,
+                                    self_attn=self_attn,
+                                    scaling=scaling,
+                                    forecast_channel_indices=forecast_channel_indices,
+                                    gated_attn=gated_attn,
+                                    loss=loss,
+                                )
+
+                                self.check_module(task="forecast", params=params)
+
+    def test_classification(self):
+        for mode in ["common_channel", "mix_channel"]:
+            for self_attn in [True, False]:
+                for scaling in [True, False, "mean", "std"]:
+                    for gated_attn in [True, False]:
+                        for head_aggregation in ["max_pool", "avg_pool"]:
+                            params = self.__class__.params.copy()
+                            params.update(
+                                mode=mode,
+                                self_attn=self_attn,
+                                scaling=scaling,
+                                head_aggregation=head_aggregation,
+                                gated_attn=gated_attn,
+                            )
+                            # print(mode,self_attn,revin,gated_attn,head_aggregation)
+
+                            self.check_module(task="classification", params=params)
+
+    def test_regression(self):
+        for mode in ["common_channel", "mix_channel"]:
+            for self_attn in [True, False]:
+                for scaling in [True, False, "mean", "std"]:
+                    for gated_attn in [True, False]:
+                        for head_aggregation in ["max_pool", "avg_pool"]:
+                            for loss in ["mse", "nll"]:
+                                params = self.__class__.params.copy()
```

Comment:
thing we should use parametrized for this

Reply:
corrected